tests/compiler/rc/logical-or-mismatch.optimized.wat

```diff
@@ -831,9 +831,42 @@
  )
  (func $~lib/rt/tlsf/growMemory (; 11 ;) (type $FUNCSIG$vii) (param $0 i32) (param $1 i32)
   (local $2 i32)
+  (local $3 i32)
+  i32.const 1
+  i32.const 27
+  local.get $1
+  i32.clz
+  i32.sub
+  i32.shl
+  i32.const 1
+  i32.sub
+  local.get $1
+  i32.add
+  local.get $1
+  local.get $1
+  i32.const 536870904
+  i32.lt_u
+  select
+  local.set $1
   memory.size
   local.tee $2
+  i32.const 16
+  i32.shl
+  i32.const 16
+  i32.sub
+  local.get $0
+  i32.load offset=1568
+  i32.eq
+  if (result i32)
```

Comment:
Hmm, that's interesting why binaryen skip `select` here and use if/else instead?

Reply:
Yeah, good question. We should certainly investigate this to understand the condition better.